src/System.Management.Automation/engine/parser/tokenizer.cs

```diff
@@ -3685,9 +3697,41 @@ private void ScanNumberAfterDot(StringBuilder sb, ref int signIndex, ref bool no
                             }
 
                             // If we're expecting a sign bit, remove the leading 0 added in ScanNumberHelper
-                            if (!suffix.HasFlag(NumberSuffixFlags.Unsigned) && ((strNum.Length - 1) & 7) == 0)
+                            if (!suffix.HasFlag(NumberSuffixFlags.Unsigned))
                             {
-                                strNum = strNum.Slice(1);
+                                var expectedLength = suffix switch
+                                {
+                                    NumberSuffixFlags.SignedByte => 2,
+                                    NumberSuffixFlags.Short => 4,
+                                    NumberSuffixFlags.Long => 16,
+                                    // No suffix flag can mean int or long depending on input string length
+                                    _ => strNum.Length < 16 ? 8 : 16
+                                };
+
+                                if (strNum.Length == expectedLength)
+                                {
+                                    strNum = strNum.Slice(1);
+                                }
+                            }
+
+
+                            // If we're expecting a sign bit, remove the leading 0 added in ScanNumberHelper
+                            if (!suffix.HasFlag(NumberSuffixFlags.Unsigned))
```

Comment:
Why is this a dup of 3700 line?

Reply:
Oh, _weird_. Guessing a copy-paste error. Thanks for noticing that! I'll edit the duplicate segment out. 💖